statsd/pipe_windows_test.go

```diff
@@ -23,32 +40,83 @@ func TestPipeWriter(t *testing.T) {
 		InputBufferSize:    1_000_000,
 	})
 	if err != nil {
-		log.Fatal(err)
+		t.Fatal(err)
 	}
 	out := make(chan string)
-	go func() {
-		conn, err := ln.Accept()
-		if err != nil {
-			log.Fatal(err)
-		}
-		buf := make([]byte, 512)
-		n, err := conn.Read(buf)
-		if err != nil {
-			log.Fatal(err)
-		}
-		conn.Close()
-		out <- string(buf[:n])
-	}()
+	go acceptOne(t, ln, out)
 
 	client, err := New(pipepath)
 	if err != nil {
-		log.Fatal(err)
+		t.Fatal(err)
 	}
 	if err := client.Gauge("metric", 1, []string{"key:val"}, 1); err != nil {
-		log.Fatal(err)
+		t.Fatal(err)
 	}
 	got := <-out
 	if exp := "metric:1|g|#key:val"; got != exp {
 		t.Fatalf("Expected %q, got %q", exp, got)
 	}
 }
+
+func TestPipeWriterReconnect(t *testing.T) {
+	f, err := ioutil.TempFile("", "test-pipe-")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.Remove(f.Name())
```

Comment:
The pipe name can be any string and does not require a file to exist unlike to Unix domain socket.